mettle/src/stdapi/net/server.c

```diff
@@ -168,7 +185,7 @@ void net_server_register_handlers(struct mettle *m)
 	struct channelmgr *cm = mettle_get_channelmgr(m);
 
 	struct channel_callbacks tcp_server_cbs = {
-		.new_cb = tcp_server_new,
+		.new_async_cb = tcp_server_new,
```

Comment:
This is now an async call back so the function itself can queue the response and have the control necessary to add the additional TLVs.